lib/pf/radius/rest.pm

```diff
@@ -0,0 +1,89 @@
+package pf::radius::rest;
+
+=head1 NAME
+
+pf::radius::rest
+
+=cut
+
+=head1 DESCRIPTION
+
+Helper methods to use REST with FreeRADIUS
+
+=cut
+
+use strict;
+use warnings;
+
+use pf::log;
+use Apache2::Const -compile =>
+  qw(DONE OK DECLINED HTTP_UNAUTHORIZED HTTP_NOT_IMPLEMENTED HTTP_UNSUPPORTED_MEDIA_TYPE HTTP_PRECONDITION_FAILED HTTP_NO_CONTENT HTTP_NOT_FOUND SERVER_ERROR HTTP_OK HTTP_INTERNAL_SERVER_ERROR);
+use pf::api::error;
+
+=head2 format_response
+
+Format a PacketFence RADIUS response to the format expected by the FreeRADIUS REST module
+
+=cut
+
+sub format_response {
+    my ($response) = @_;
+
+    my $radius_return = shift @$response; 
+    my %mapped_object = @$response; 
+    %mapped_object = ( %{$mapped_object{"RADIUS_AUDIT"}}, %mapped_object);
+
+    delete $mapped_object{"RADIUS_AUDIT"};
+    use Data::Dumper; get_logger->debug(sub { use Data::Dumper ; "RADIUS REST object : ". Dumper(\%mapped_object) });
+    $response = \%mapped_object; 
+    
+    unless ($radius_return == 2) { 
+        die pf::api::error->new(status => Apache2::Const::HTTP_UNAUTHORIZED, response => $response);
+    }
+
+    return $response;
+}
+
+=head2 format_request
+
+Format a FreeRADIUS REST RADIUS request to the format PacketFence expects it
+
+=cut
+
+sub format_request {
+    my ($request) = @_;
+    # transform the request according to what radius_authorize expects
+    my %remapped_radius_request = map {
+        $_ => $request->{$_}->{value}->[0];
```

Comment:
Just use the first.
Joining them would break as many things as using a list in the first place.

I know it's not ideal, but we have lived with it for a long time anyway.
There was always the risk that the NAS could send multiple values.
We never dealt with them properly before either.